homeassistant/components/transmission/switch.py

```diff
@@ -70,20 +77,20 @@ def available(self):
 
     def turn_on(self, **kwargs):
         """Turn the device on."""
-        if self.type == "on_off":
+        if self.entity_description.key == "on_off":
             _LOGGING.debug("Starting all torrents")
             self._tm_client.api.start_torrents()
-        elif self.type == "turtle_mode":
+        elif self.entity_description.key == "turtle_mode":
             _LOGGING.debug("Turning Turtle Mode of Transmission on")
             self._tm_client.api.set_alt_speed_enabled(True)
```

Comment:
We can add an `on_func` to the description to call the correct method.
Refer to the below example.

https://github.com/engrbm87/home-assistant-core/blob/4a938ec33ee8df17e619fc64a519ef7c2ff856d8/homeassistant/components/android_ip_webcam/switch.py#L20-L42

Reply:
Funny how I just wanted to add device_info in this PR and ended up doing a big refactor.